### ISSUE TITLE
update 'print_to_file' fortran benchmark

### ISSUE THREAD

**Issue by gha3mi** [JuliaLang/Microbenchmarks#70]
The use of implied DO-loops leads to a better performance.

**Comment by zoziha** [comment]
Yes, an implicit do loop (for [perf.f90](https://github.com/JuliaLang/Microbenchmarks/blob/2f81eea6b6999e2480d42f8bde77a46b3c4b80e8/perf.f90#L220)) is more efficient:
```fortran
subroutine printfd(n)
integer, intent(in) :: n
integer :: i
character(*), parameter :: newline = new_line("")
open(unit=1, file="/dev/null")
write(unit=1, fmt=*) (i, i+1, newline, i=1, n)
close(unit=1)
end subroutine
```
`newline` is a carriage return.
Maybe we can create a PR for this.

**Comment by gha3mi** [comment]
Thanks @zoziha. I have checked your version with gfortran and ifort. Could you please create a PR?

Using GNU Fortran (GCC) 12.1.0: `gfortran perf.f90`
| Previous Version      | New Version |
| ----------- | ----------- |
| 23.536152      | 9.529248       |

Using ifort (IFORT) 2021.5.0 20211109: `ifort perf.f90`
| Previous Version      | New Version |
| ----------- | ----------- |
| 31.675000      | 14.640000      |
